app/src/main/java/org/mifos/mobile/ui/fragments/NewHomeFragment.kt

```diff
@@ -0,0 +1,104 @@
+package org.mifos.mobile.ui.fragments
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import kotlinx.android.synthetic.main.content_fragment_home.*
+import kotlinx.android.synthetic.main.fragment_home.*
+import org.mifos.mobile.R
+import org.mifos.mobile.ui.activities.NotificationActivity
+import org.mifos.mobile.ui.fragments.base.BaseFragment
+
+class NewHomeFragment : BaseFragment(), View.OnClickListener {
+
+    private lateinit var rootView: View
+    private var currentSelectedButton = 1;
+
+    companion object {
+        fun newInstance(): NewHomeFragment = NewHomeFragment()
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        rootView =  inflater.inflate(R.layout.fragment_home, container, false)
+
+        val ft = childFragmentManager.beginTransaction()
+        val rt = RecentTransactionsFragment()
+        ft.replace(R.id.fl_recentTransactions, rt)
+        ft.addToBackStack(null)
+        ft.commit()
+
+        return rootView
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        iv_notification.setOnClickListener(this)
+        btn_all.setOnClickListener(this)
+        btn_third_party.setOnClickListener(this)
+        btn_self.setOnClickListener(this)
+    }
+
+    override fun onClick(view: View?) {
+        when(view?.id){
+            R.id.iv_notification -> {
+                startActivity(Intent(context, NotificationActivity::class.java))
+            }
+
+            R.id.btn_all -> {
+                if (currentSelectedButton != 1){
+                    currentSelectedButton = 1
```

Comment:
`currentSelectedButton` can we a method variable (updateUI), not needed to have it as a class variable. check other places also

Reply:
Actually this variable is used inside another function `onClick`. So, I think we need this as a class variable.